io/io/src/RRawFileCreate.cxx

```diff
@@ -0,0 +1,38 @@
+// @(#)root/io:$Id$
+// Author: Jakob Blomer
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <ROOT/RConfig.h>
+#include <ROOT/RRawFile.hxx>
+
+#include "TPluginManager.h"
+#include "TROOT.h"
+
+#include <memory>
+
+std::unique_ptr<ROOT::Internal::RRawFile> ROOT::Internal::RRawFile::CreateHttpRRawFile(std::string_view url, ROptions options)
+{  
+   std::string transport = GetTransport(url);
+   std::unique_ptr<RRawFile>  helper;
+   if (transport == "http" || transport == "https") {
+      if (TPluginHandler *h = gROOT->GetPluginManager()->FindHandler("ROOT::Internal::RRawFile")) {
+         if (h->LoadPlugin() == 0) {
+            //return std::unique_ptr<RRawFile>(reinterpret_cast<RRawFile *>(h->ExecPlugin(2, &url, &options)));
+            helper = std::unique_ptr<RRawFile>(reinterpret_cast<RRawFile *>(h->ExecPlugin(2, &url, &options)));
+            ROOT::Internal::RRawFile::InitHelper(std::move(helper));
```

Comment:
This is not the intent.  This would provide to RRawFile is *single* instance of a http helper.

Instead the function implement here `ROOT::Internal::RRawFile::CreateHttpRRawFile` needs to have its address passed on to the InitHelper.  

Its declaration also needs to move out of `RRawFile` otherwise the mini library would have an unresolved symbol.  So the code might look more like:
```
std::unique_ptr<ROOT::Internal::RRawFile> ROOT::Internal::CreateHttpRRawFile(std::string_view url, ROptions options)
{
.... /* no call to InitHelper *./
};
```
and
```
static int gBootstrapingStatic = ROOT::Internal::RRawFile::InitHelper(& ROOT::Internal::RRawFile::CreateHttpRRawFile)
```

PS (Another alternative  is to define an abstract Factory interface in the mini library and to have the derived "complete" factory be registered in a way similar to the one mentioned here.